app/views/questionnaire.py

```diff
@@ -170,43 +173,39 @@ def post_household_composition(eq_id, form_type, collection_id, group_id):  # py
 
 
 @questionnaire_blueprint.route('thank-you', methods=["GET"])
-@login_required
 def get_thank_you(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
-    metadata = get_metadata(current_user)
-    if metadata.get('submitted', False):
-        theme = g.schema_json.get('theme', None)
-        logger.debug("theme selected", theme=theme)
-        metadata = get_metadata(current_user)
-        metadata_context = build_metadata_context(metadata)
-        thank_you_template = render_theme_template(theme, template_name='thank-you.html',
+    survey_completed_metadata = session_storage.get_survey_completed_metadata()
+    schema = schema_loader.load_schema(eq_id, form_type)
+
+    if survey_completed_metadata:
+        metadata_context = build_metadata_context_for_survey_completed(survey_completed_metadata)
+        thank_you_template = render_theme_template(schema['theme'], template_name='thank-you.html',
                                                    meta=metadata_context,
-                                                   legal_basis=g.schema_json['legal_basis'],
                                                    analytics_ua_id=settings.EQ_UA_ID,
-                                                   survey_id=g.schema_json['survey_id'],
-                                                   survey_title=TemplateRenderer.safe_content(g.schema_json['title']))
-        _delete_user_data()
+                                                   survey_id=schema['survey_id'],
+                                                   survey_title=TemplateRenderer.safe_content(schema['title']))
         return thank_you_template
     else:
-        answer_store = get_answer_store(current_user)
-        path_finder = PathFinder(g.schema_json, answer_store, metadata)
-        full_routing_path = path_finder.get_routing_path()
-        latest_location = path_finder.get_latest_location(get_completed_blocks(current_user),
-                                                          routing_path=full_routing_path)
-        return _redirect_to_latest_location(collection_id, eq_id, form_type, latest_location)
+        return get_latest_location(collection_id, eq_id, form_type, schema)
 
 
-@questionnaire_blueprint.route('signed-out', methods=["GET"])
 @login_required
+def get_latest_location(collection_id, eq_id, form_type, schema):
```

Comment:
maybe this should be `_redirect_to_latest_location` and the original `_redirect_to_latest_location` should be renamed `_redirect_to_location` as it just take a `location` anyway?